packages/use-dataloader/src/useDataLoader.ts

```diff
@@ -1,171 +1,126 @@
-import {
-  useCallback,
-  useEffect,
-  useLayoutEffect,
-  useMemo,
-  useRef,
-  useState,
-} from 'react'
+import { useCallback, useEffect, useRef, useState } from 'react'
 import { useDataLoaderContext } from './DataLoaderProvider'
 import { StatusEnum } from './constants'
+import DataLoader from './dataloader'
 import { PromiseType, UseDataLoaderConfig, UseDataLoaderResult } from './types'
 
-/**
- * @param {string} key key to save the data fetched in a local cache
- * @param {() => PromiseType} method a method that return a promise
- * @param {useDataLoaderConfig} config hook configuration
- * @returns {useDataLoaderResult} hook result containing data, request state, and method to reload the data
- */
-const useDataLoader = <T>(
+function useDataLoaderV2<ResultType, ErrorType = Error>(
```

Comment:
Why renaming in v2 ? Anyways we have to migrate all old ones as there is no coexisting v1 and v2 ?